### ISSUE TITLE
swN2() demands first arg be CTD, if 'eos' given

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1524]
This came up in fixing #1523 

```R
argo[["N2"]]
 Hide Traceback
 
 Rerun with Debug
 Error in swN2(pressure, sigmaTheta = x[["sigmaTheta"]], eos = getOption("oceEOS",  : 
  first argument must be a CTD object if eos="gsw" 
4.
stop("first argument must be a CTD object if eos=\"gsw\"") at sw.R#323
3.
swN2(pressure, sigmaTheta = x[["sigmaTheta"]], eos = getOption("oceEOS", 
    default = "gsw")) at argo.R#153
2.
argo[["N2"]] 
1.
argo[["N2"]] 
```

**Comment by dankelley** [comment]
(Question for @richardsc at the end.)

Since the `eos="gsw"` formulation uses `SA` and `CT`, as opposed to `sigmaTheta`, fixing this is a bit tricky, given the present argument list, which does not mention `SA` and `CT`, or `salinity`+`temperature`+`longitude`+`latitude`, which are required compute `SA` and `CT`.

This `swN2` function is *old* and I want to think some more on how to fix it.

As for #1523, i.e. making `argo[["N2"]]` work properly, we can get that without forcing `swN2()` changes that have not come up in years of usage.  So that means the pressure is off, in terms of addressing the present issue.

Frankly, I do not trust the `gsw` methodology, which does no smoothing, as much as the one we have already, and calling that present thing `eos="unesco"` is not really informative.  My present thought is simply to **remove** the `eos` argument.  What do you think of that, @richardsc ?

**Comment by richardsc** [comment]
I agree with removing the `eos` arg from `swN2()`. It was never really a "unesco" function anyway, and if someone wants gsw N2 they can use the gsw function directly (forget what it's called).

**Comment by dankelley** [comment]
Fixed in "develop" branch, commit 7942a3a2b3a007f2bd2d38eef51c70778244135c
